aws/data_source_aws_ec2_instance_type.go

```diff
@@ -0,0 +1,440 @@
+			},
+			"ena_support": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"gpus": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_gpu_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"fpgas": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_fpga_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"supported_placement_strategies": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"accelerators": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"hibernation_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"burstable_performance_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"dedicated_hosts_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"auto_recovery_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEc2InstanceTypeRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	params := &ec2.DescribeInstanceTypesInput{}
+
+	if v, ok := d.GetOk("instance_type"); ok {
+		params.InstanceTypes = []*string{aws.String(v.(string))}
+	}
+	log.Printf("[DEBUG] Reading instances types: %s", params)
+
+	resp, err := conn.DescribeInstanceTypes(params)
+	if err != nil {
+		return err
+	}
+	if len(resp.InstanceTypes) == 0 {
+		return fmt.Errorf("no EC2 Instance Type found matching criteria; try different search")
+	}
+	v := resp.InstanceTypes[0]
```

Comment:
It seems unlikely that it would return more than one. However, if the API shifts and we're just grabbing the first, no one will know multiple were returned. Let's give an error.

```suggestion
	if len(resp.InstanceTypes) > 1 {
 		return fmt.Errorf("multiple instance types found for type %s", instanceType)
 	}
	v := resp.InstanceTypes[0]
```